modules/tinymce/src/plugins/link/main/ts/core/Actions.ts

```diff
@@ -6,9 +6,16 @@ import VK from 'tinymce/core/api/util/VK';
 import * as OpenUrl from './OpenUrl';
 import * as Utils from './Utils';
 
-const getLinks = (editor: Editor) => editor.selection.isCollapsed() ?
-  Utils.getLinks(editor.dom.getParents(editor.selection.getStart())) as [HTMLAnchorElement] :
-  Utils.getLinksInSelection(editor.selection.getRng());
+const isSelectionOnImageWithEmbeddedLink = (editor: Editor) => {
+  const rng = editor.selection.getRng();
+  const node = rng.startContainer;
+  // Handle a case where an image embedded with a link is selected
+  return Utils.isLink(node) && rng.startContainer === rng.endContainer && editor.dom.select('img', node).length === 1;
+};
+
+const getLinks = (editor: Editor) => editor.selection.isCollapsed() || isSelectionOnImageWithEmbeddedLink(editor)
```

Comment:
I am wondering if we can keep this more generic so that it covers any selectable node that is wrapped in an `a` tag

```suggestion
const getLinks = (editor: Editor) => editor.selection.isCollapsed() || (editor.selection.getStart() === editor.selection.getEnd())
```

Reply:
It would fail a test case where it executes open link action on a link inside selection. In that case, `getLinks` should trigger `Utils.getLinksInSelection` instead